src/Orleans/Storage/RelationalConstants.cs

```diff
@@ -0,0 +1,379 @@
+OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+*/
+
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Data;
+using System.Data.Common;
+using System.Diagnostics;
+using System.Linq;
+
+
+namespace Orleans.Runtime.Storage.Relational
+{
+    /// <summary>
+    /// A holder for well known, vendor specific connector class invariant names.
+    /// </summary>
+    public static class WellKnownRelationalInvariants
+    {
+        /// <summary>
+        /// Microsoft SQL Server.
+        /// </summary>
+        public const string SqlServer = "System.Data.SqlClient";
+
+        /// <summary>
+        /// Oracle Database server.
+        /// </summary>
+        public const string OracleDatabase = "Oracle.DataAccess.Client";
+    }
+
+
+    /// <summary>
+    /// Constants Orleans uses to relational database management.
+    /// </summary>
+    [DebuggerDisplay("InvariantName = {InvariantName}, AssemblyQualifiedName = {AssemblyQualifiedName}")]
+    public class RelationalConstants
+    {
+        /// <summary>
+        /// A key for a well known constant, likely set by a database connector vendor.
+        /// </summary>
+        public const string NameKey = "Name";
+
+        /// <summary>
+        /// A key for a well known constant, likey set by a database connector vendor.
+        /// </summary>
+        public const string DescriptionKey = "Description";
+
+        /// <summary>
+        /// A key for a well known constant, likely set by a database connector vendor.
+        /// </summary>
+        public const string InvariantNameKey = "InvariantName";
+
+        /// <summary>
+        /// A key for a well known constant, likely set by a database connector vendor.
+        /// </summary>
+        public const string AssemblyQualifiedNameKey = "AssemblyQualifiedName";
+
+        /// <summary>
+        /// A key to add a default connection string if none otherwise defined.
+        /// This is something a given database will be listening to. If not altered during
+        /// setup, usually databases have well known default settings.
+        /// </summary>
+        /// <remarks>Vendor specific.</remarks>
+        public const string DefaultConnectionStringKey = "DefaultConnectionString";
+
+        /// <summary>
+        /// The character that indicates a parameter.
+        /// </summary>
+        /// <remarks>Vendor specific.</remarks>
+        public const string ParameterIndicatorKey = "ParameterIndicator";
+
+        /// <summary>
+        /// The character that indicates a start escape key for columns and tables that are reserved words.
+        /// </summary>
+        /// <remarks>Vendor specific.</remarks>
+        public const string StartEscapeIndicatorKey = "StartEscapeIndicator";
```

Comment:
For SQL Server these are square brackets, for MySQL these are backticks and so forth for other databases. These are used if/when queries need to be ran on tables that have reserved words are column names. More to do with general ORM specific functionality (though Orleans uses some reserved words too).